gtda/diagrams/features.py

```diff
@@ -447,8 +458,9 @@ class HeatKernel(BaseEstimator, TransformerMixin):
 
     """
 
-    _hyperparameters = {'sigma': [numbers.Number, (1e-16, np.inf)],
-                        'n_bins': [int, (1, np.inf)]}
+    _hyperparameters = {
+        'n_bins': {'type': int, 'in': Interval(1, np.inf, closed='left')},
+        'sigma': {'type': Real, 'in': Interval(0, np.inf, closed='neither')}}
```

Comment:
I'd think one is likely to get warnings from `scipy` in that case. For me it boils down to whether we are we likely to have a better idea than them about this. The previous limits were there because we could only support closed intervals and wanted to exclude `0.` cheaply.